### PR TITLE
Finished

### PR DESCRIPTION
**Pull request by joyydong**
## In this PR
Web app to search for food recipes, made with React.

### 🖋  What I learned

- React syntax and implementation (useState, useEffect)
- How to write Javascript and HTML within React code
- Using CSS modules for custom styling
- Checking log statements in Inspect Element as a form of verification

### 🤳 Screenshots (if applicable)

Initial page—set to banana recipes as default
image

Searching for noodle recipes
image

### 🐞 Questions

- How does bp feel about disabling eslint-rules? I fixed a `useEffect missing dependency warning` by adding the method into `useEffect` ([cite](https://stackoverflow.com/questions/55840294/how-to-fix-missing-dependency-warning-when-using-useeffect-react-hook)) but noticed that disabling the eslint-rule was also an option, which might be better in case we need to use `getRecipe` elsewhere. 
- I personally add a lot of comments to my code to help me understand and improve readability for others. In the future, is there a specific format I should follow?

### CODE REVIEW

**Comment by micahtyong** [comment]
Great work on this Joy! Going to pass this along to Ethan, your onboarding mentor, who can help answer some of your questions.

**Comment by ethanlee16** [comment]
> How does bp feel about disabling eslint-rules? I fixed a useEffect missing dependency warning by adding the method into useEffect (cite) but noticed that disabling the eslint-rule was also an option, which might be better in case we need to use getRecipe elsewhere.

Great question — definitely feel this as someone who has found myself unproductively fighting the linter before and also as someone who believes that stylistic lint rules are mostly disable-able.

However, the exhaustive-deps lint rule in particular _is_ important and could make a functional difference in your code! Let's take a look at the effect in question:

```tsx
useEffect( () => {
  const getRecipes = async () => {
    // can also use .then
    const response = await fetch(`https://api.edamam.com/search?q=${query}&app_id=${APP_ID}&app_key=${APP_KEY}`); // external request
    const data = await response.json(); // need to wait for response to come 
    setRecipes(data.hits);
    console.log(data.hits);
  };
  getRecipes();
}, [query]); // only runs when we click submit bc that's the only time query will change
```

If we remove `query` from the dependencies array (second arg of the useEffect call), this means that the effect code won't update in response to changes to `query`. That means that `getRecipes` will only be called _once,_ after the first mount of the component (this is the meaning of an empty dependency array) and the `query` value within `getRecipes` will always refer to the first value of "chicken".

It sounds like you had initially tried making `getRecipes` outside of the effect - _love_ that you're thinking about reusability! I'm not 100% sure how you had tried this, but the "hooks" approach to reusing this logic might look something like:

```tsx
/*
 * Custom hook to fetch recipes by a query string.
 */
export const useRecipes = (query) => {
  // Set up stateful values that will be used for the output to this hook
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [recipes, setRecipes] = useState([]);

  useEffect(() => {
      async getRecipes() {
         try {
           setRecipes(await fetch(...));
         } catch (err) {
           setError(err);
         }
         setLoading(false);
      }
      getRecipes();
  }, [query]);

  return { loading, error, recipes };
}
```

Notice how we still include the `getRecipes` function within the effect, and that effect still has `query` as a dependency! We could probably extract out the fetch pieces of the `getRecipes` function if just the request (without the React-specific logic) needed to be reused elsewhere.

_(One constructive comment as a tangent to the above: if you have an initial approach that you make some progress on, I'd recommend committing that approach early, even if it doesn't quite work yet. It can be helpful to have those snapshots for others to see the "graph" of your work to see what you had tried!)_

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,59 @@
...
+  const [search, setSearch] = useState(' '); // search state
+  const [query, setQuery] = useState('chicken'); // search terms we are querying by
+
+  useEffect( () => {
+    const getRecipes = async () => {
+      // can also use .then
+      const response = await fetch(`https://api.edamam.com/search?q=${query}&app_id=${APP_ID}&app_key=${APP_KEY}`); // external request
+      const data = await response.json(); // need to wait for response to come 
+      setRecipes(data.hits);
+      console.log(data.hits);
```
Comment (erinysong-zz) [inline src/App.js:19]: nice job using console.log to debug! for future practice, it's generally good to clean up your commits by remove all your console.log statements once you're done using them to debut

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,59 @@
...
+        <button className="search-button" type="submit">
+          Search
+        </button>
+      </form>
+      <div className="recipes">
+        {recipes.map(recipe => (
+          <Recipe 
+            key={recipe.recipe.label} // Each child in a list should have a unique "key" prop
+            title={recipe.recipe.label} 
+            calories={Math.round(recipe.recipe.calories)}
```
Comment (erinysong-zz) [inline src/App.js:49]: good job rounding the calorie data to the nearest integer! it makes the data much more readable and user friendly :^)

**Review by erinysong-zz — COMMENTED** [review]
awesome job joy! you did a really good job using React Hooks to update the state of your app and I appreciate the time you took to style everything to be more user friendly (e.g., labeling what the calories number represented, rounding the calorie quantity, image padding). super excited to build with you this semester! <3

File: src/App.js (hunk 1 above)
Comment (joyydong) [reply src/App.js:19]: got it! i noticed that I forgot to delete the console.log line yesterday too while looking over the code with Ethan, but will do in the future :)

**Pull request merged**